### PR TITLE
Add Via keymap for BM16-A

### PR DESCRIPTION
**Pull request by epid**
## Description

Added VIA keymap for the KPRepublic BM16-A

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bm16a/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum layers {
+  _BASE = 0,
```
Comment (drashna) [inline keyboards/bm16a/keymaps/via/keymap.c:19]: Just a heads up, you don't need the ` = 0` part here.  It starts at 0, but it's fine.

File: keyboards/bm16a/config.h [hunk 2]
```diff
@@ -19,8 +19,8 @@
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
-#define PRODUCT_ID      0x424D
+#define VENDOR_ID       0x4B50  // Adding via support
```
Comment (noroadsleft) [inline keyboards/bm16a/config.h:22]:
```suggestion
#define VENDOR_ID       0x4B50  // "KP"
```

File: keyboards/bm16a/keymaps/via/config.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2019
```
Comment (noroadsleft) [inline keyboards/bm16a/keymaps/via/config.h:1]: This file should be deleted.

File: keyboards/bm16a/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS, \
+    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS, \
+    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS  \
+  ),
+
+  [3] = LAYOUT_ortho_4x4(
+    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS, \
+    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS, \
+    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS, \
+    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS  \
```
Comment (fauxpark) [inline keyboards/bm16a/keymaps/via/keymap.c:49]:
```suggestion
    KC_PGUP, KC_HOME,  KC_UP,    KC_END ,
    KC_PGDN, KC_LEFT,  KC_DOWN,  KC_RGHT,
    KC_DOT,  KC_VOLU,  KC_MPLY,  KC_MPRV,
    MO(1),  KC_VOLD,  KC_MUTE,  KC_MNXT
  ),
  [1] = LAYOUT_ortho_4x4(
    RESET,   KC_TRNS,  KC_TRNS,  KC_TRNS,
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS
  ),
  [2] = LAYOUT_ortho_4x4(
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS
  ),

  [3] = LAYOUT_ortho_4x4(
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,
    KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS
```

**Comment by Erovia** [comment]
Thank you for your contribution!

**Pull request merged**